commit 90e493d7d51c791a2adc2de962cbd5000f1b7460
Author: Thor Thayer <tthayer@opensource.altera.com>
Date:   Wed Oct 19 14:53:47 2016 -0500

    EDAC, altera: Disable IRQs while injecting SDRAM errors
    
    Disable IRQs while injecting SDRAM errors. The RT patches exposed
    a spinlock deadlock where the spinlock taken for the regmap write
    deadlocked with the IRQ clear regmap write.
    
    Error injection is not normally enabled for ECC but only for testing.
    
    Signed-off-by: Thor Thayer <tthayer@opensource.altera.com>
    Cc: linux-edac <linux-edac@vger.kernel.org>
    Link: http://lkml.kernel.org/r/1476906827-9412-1-git-send-email-tthayer@opensource.altera.com
    Signed-off-by: Borislav Petkov <bp@suse.de>

diff --git a/drivers/edac/altera_edac.c b/drivers/edac/altera_edac.c
index 58d3e2b39b5b..6421cc3c7dc1 100644
--- a/drivers/edac/altera_edac.c
+++ b/drivers/edac/altera_edac.c
@@ -153,13 +153,17 @@ static ssize_t altr_sdr_mc_err_inject_write(struct file *file,
 	if (count == 3) {
 		edac_printk(KERN_ALERT, EDAC_MC,
 			    "Inject Double bit error\n");
+		local_irq_disable();
 		regmap_write(drvdata->mc_vbase, priv->ce_ue_trgr_offset,
 			     (read_reg | priv->ue_set_mask));
+		local_irq_enable();
 	} else {
 		edac_printk(KERN_ALERT, EDAC_MC,
 			    "Inject Single bit error\n");
+		local_irq_disable();
 		regmap_write(drvdata->mc_vbase,	priv->ce_ue_trgr_offset,
 			     (read_reg | priv->ce_set_mask));
+		local_irq_enable();
 	}
 
 	ptemp[0] = 0x5A5A5A5A;